e2e/test-helpers.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/exec"
+)
+
+func expectActions(got []exec.Action, want []exec.ActionType) error {
+	var errs []error
+WantLoop:
+	for _, wantT := range want {
```

Comment:
If you do this, if there are extra in `got`, you wouldn't know. This is a bug.

If order does not matter, I suggest putting into a set or map and then calling `cmp.Diff` instead of trying to write custom loops like this.

Reply:
the idea was to check if got contains want not if they are equal. 
But actually we can check all actions

Reply:
changed, I needed to update existing test and add all actions including events